tcms/static/js/utils.js

```diff
@@ -1,17 +1,16 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
-    // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
-        new_options = _select_tag.options[0].outerHTML;
-    }
-
+	currentGroup = '';
     data.forEach(function(element) {
+		if (currentGroup !== element[group_attr] && group_attr != null) {
+			new_options += '<optgroup label="' + element[group_attr] + '">';
```

Comment:
I think this for loop should also be conditional based on whether or not `_select_tag` has the multiple attribute defined. On some pages we want single selection and it doesn't make sense to add groups because there is only 1 group anyway.  For example on /runs/new/ the Test Plan selector is a single select b/c you can't create a test run attached to multiple test plans.

Reply:
The `updateSelect` function is used in several other places which depending on the page will allow or will not allow multiple select which also reinforces the comment above.

Reply:
Done